### PR TITLE
Generate unique event IDs inside the event based rupture calculator

### PR DESCRIPTION
**Pull request by micheles**
See the discussion in https://github.com/gem/oq-engine/issues/2527. One big advantage of this approach is that the IDs are reproducible once `concurrent_tasks` is set (i.e. if you run the same computation twice you get the same event IDs).

The algorithm used to build the event ID has a limit of  65,536 tasks, and each task can generate at most 65,536 ruptures. All known event based calculations are well below such limits; if in the future we will pass the limits, it will be trivial to raise them (change the `eid` from a 32 bit unsigned integer to a 64 bit unsigned integer: then the limits will be 4,294,967,296 tasks each one producing at most  4,294,967,296 ruptures).

The demos are running here: https://ci.openquake.org/job/zdevel_oq-engine/2390/

### CODE REVIEW

File: openquake/calculators/event_based.py [hunk 1]
```diff
@@ -144,9 +166,8 @@ def build_eb_ruptures(
                 num_occ_by_rup[rup].items()):
             for occ_no in range(1, num_occ + 1):
                 # NB: the eid below is a placeholder; the right eid will be
-                # set later, in EventBasedRuptureCalculator.post_execute
-                events.append((eid, ses_idx, occ_no, sampleid))
-                eid += 1
+                # set a but later, in compute_ruptures
```
Comment (daniviga) [inline openquake/calculators/event_based.py]: but -> bit?

File: openquake/commonlib/logs.py [hunk 2]
```diff
@@ -15,17 +15,18 @@
...
 # along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
-
-
 """
 Set up some system-wide loggers
 """
 
 import os.path
 import logging
+import multiprocessing
```
Comment (daniviga) [inline openquake/commonlib/logs.py]: Why are you importing multiprocessing which is not directly used?

File: openquake/commonlib/logs.py [hunk 3]
```diff
@@ -15,17 +15,18 @@
...
 from datetime import datetime
 from contextlib import contextmanager
 from multiprocessing.connection import Client
+import numpy
```
Comment (daniviga) [inline openquake/commonlib/logs.py]: idem (and in any case I would group it with 'imports' before 'froms')

File: openquake/commonlib/logs.py (hunk 2 above)
Comment (micheles) [inline openquake/commonlib/logs.py]: It's a leftover from a past implementation, I am removing it

File: openquake/calculators/event_based.py (hunk 1 above)
Comment (micheles) [inline openquake/calculators/event_based.py]: yes, typo

**Pull request merged**